distribution/tools/entitlement-runtime/src/main/java/org/elasticsearch/entitlement/runtime/policy/EntitlementBuilder.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.entitlement.runtime.policy;
+
+import org.elasticsearch.xcontent.XContentParser;
+
+import java.io.IOException;
+
+public abstract class EntitlementBuilder {
```

Comment:
I wonder, given the multi-phase parse-validate-build approach you mentioned in the comments... should this be called `EntitlementParser`?

Reply:
++ (I suggested the same, Parser or Reader)

Reply:
I renamed everything to parser. I had considered these builders for the validation/build step, but maybe what makes the most sense is to separate out the builders completely from the parsers. So it would end up being parser -> builder -> immutable policy object(s). If that's the case I'll have the parsers return builder objects. WDYT?

Reply:
FWIW my hunch is that the extra step will pay off. There's a tendency to make parsers do a lot, but unless they're very simple, parsers are best left just determining what the text _says_ rather than trying to validate that it makes sense, or trying to "execute" it by building some sort of efficient data structures.

It makes sense to parse into some convenient form, validate that, and then once we have all the validated grants, run a build step that generates a highly efficient data structure for entitlement checks.

It's not super clear though, because it's possible our situation will be simple enough that this distinction isn't pulling its weight, and the parser could just generate efficient data structures directly. 🤷